[WIP] Добавить CancellationToken в асинхронные методы

itsokto opened the pull request.

itsokto committed:
Добавить асинхронные методы в ICaptchaSolver

itsokto committed:
Добавить CancellationToken в IRestClient

itsokto committed:
Добавить асинхронные методы в ICaptchaHandler

itsokto committed:
Добавить CancellationToken в IVkApiInvoke

itsokto committed:
Добавить CancellationToken в IVkLongPollInvoke

itsokto committed:
Добавить CancellationToken в WebCall

itsokto committed:
Добавить CancellationToken в INeedValidationHandler

itsokto committed:
Добавить CancellationToken в IAuthorizationForm

itsokto committed:
Добавить CancellationToken в IAuthorizationFormHtmlParser

itsokto committed:
Добавить CancellationToken в IAuthorizationFlow

itsokto committed:
Добавить CancellationToken в IVkApiAuthAsync

itsokto committed:
Добавить CancellationToken в тестах

itsokto committed:
Добавить CancellationToken в AccountCategory

itsokto committed:
Пометить некоторые тесты атрибутом Ignore

Тесты падают, потому что была убрана валидация параметров из кода

mokeev1995 commented on VkNet/Categories/AccountCategory.cs:
```diff
@@ -311,10 +269,10 @@ public long GetAppPermissions(long userId)
 			};
 
-			return _vk.Call(methodName: "account.getAppPermissions", parameters: parameters);
+			return _vk.Call("account.getAppPermissions", parameters);
```
тут точно так должно быть, а не вызов асинхронного метода?

itsokto committed:
Вызывать асинхронный метод в GetAppPermissions

itsokto committed:
Добавить CancellationToken в LikesCategory

itsokto committed:
Добавить CancellationToken в ExecuteCategory

itsokto committed:
Добавить CancellationToken в AuthCategory

itsokto committed:
Вернуть явное приведение к типу User в GetProfileInfoAsync

itsokto committed:
Добавить CancellationToken в AppsCategory

itsokto committed:
Добавить CancellationToken в AppWidgetsCategory

itsokto committed:
Добавить CancellationToken в WidgetsCategory

itsokto committed:
Добавить CancellationToken в StreamingCategory

itsokto committed:
Добавить CancellationToken в StorageCategory

itsokto committed:
Добавить CancellationToken в StatusCategory

itsokto committed:
Добавить CancellationToken в StatsCategory

itsokto committed:
Добавить CancellationToken в SearchCategory

itsokto committed:
Добавить CancellationToken в NotificationsCategory

itsokto committed:
Рефакторинг LikesCategory

itsokto committed:
Добавить CancellationToken в GiftsCategory

itsokto committed:
Добавить CancellationToken в WallCategory

itsokto committed:
Добавить WallGetObjectJsonConverter

itsokto committed:
Тесты для WallGetObjectJsonConverter

itsokto committed:
Игнорировать тест GetById_IncorrectParameters_ThrowException

Причина: убрана валидация параметров метода wall.getById из кода.

itsokto committed:
Использовать null-conditional operator для доступа к свойству Attachments через свойство Attachment

itsokto committed:
Пометить Photo свойтва объекта Group атрибутом JsonProperty

itsokto committed:
Пометить Thumb свойтва объекта Gift атрибутом JsonProperty

itsokto committed:
Исправить падение теста GetPosts_Extended

itsokto committed:
Добавить публичное свойство (JToken) Token

itsokto committed:
Выполнять десериализацию объекта из VkResponse.Token

itsokto committed:
VkCollectionJsonConverter: Реализовать поиск поля массива элементов вместо его явной установки

itsokto committed:
VkCollectionJsonConverter: WriteJson должен возвращать json валидный для обратной десериализации

itsokto committed:
Тесты для VkCollectionJsonConverter

itsokto committed:
Добавить CancellationToken в MessagesCategory

itsokto committed:
Переписать тесты для метода messages.send проверяющие параметр RandomId

itsokto committed:
#831 Удалить obsolete методы MessagesCategory

itsokto committed:
LastActivity: Добавить атрибут JsonProperty над свойством IsOnline

itsokto committed:
Message: Изменить тип свойства Out на MessageType

itsokto committed:
Исправить JsonSerializationException при вызове метода messages.getChatAsync

itsokto committed:
LongPollHistoryResponse: изменить тип свойства Messages на VkCollection<Message>

itsokto committed:
Merge branch 'develop' of https://github.com/vknet/vk into fix/815

itsokto committed:
Изменить название папки с данными для тестов Newsfeed в NewsFeed

Некоторые тесты категории NewsFeed падали с ошибкой FileNotFoundException на Linux, потому что путь к папкам был указан разный, а Linux чувствителен к регистру.

mokeev1995 commented:
@inyutin-maxim хоть б написал, за что ты так с ним)

inyutin-maxim commented:
> @inyutin-maxim хоть б написал, за что ты так с ним)

Ахах, я тоже самое делаю в другой ветке. а эту развивать не планируют)
